src/lib/serviceRoutinesV2/postRegistration.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+*
+* Copyright 2018 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Ken Zangelin
+*/
+#include <string>
+#include <vector>
+
+#include "logMsg/logMsg.h"
+
+#include "common/defaultValues.h"
+#include "common/statistics.h"
+#include "common/clockFunctions.h"
+#include "common/string.h"
+#include "rest/ConnectionInfo.h"
+#include "rest/OrionError.h"
+#include "ngsi/ParseData.h"
+#include "apiTypesV2/Registration.h"
+#include "mongoBackend/mongoRegistrationCreate.h"
+#include "serviceRoutinesV2/postRegistration.h"
+#include "alarmMgr/alarmMgr.h"
+
+
+
+/* ****************************************************************************
+*
+* postRegistration - 
+*
+* POST /v2/registrations
+*
+* Payload In:  ngsiv2::Registration
+* Payload Out: None
+*/
+std::string postRegistration
+(
+  ConnectionInfo*            ciP,
+  int                        components,
+  std::vector<std::string>&  compV,
+  ParseData*                 parseDataP
+)
+{
+  ngsiv2::Registration  registration;
+  OrionError            oe;
+  std::string           servicePath = (ciP->servicePathV[0] == "")? SERVICE_PATH_ROOT : ciP->servicePathV[0];
```

Comment:
This is the right point where what I said at https://github.com/apinf/fiware-orion/pull/14/files#r162019236 applies. SERVICE_PATH_ROOT has to be changed by SERVICE_PATH_ALL.

Reply:
Postponed to after this discussion has taken place.
See issue #3078

Reply:
FIXME mark pointing to #3078 missing?

Reply:
Sure, fixed in 6f04a860e